src/Angor/Shared/Protocol/FounderTransactionActions.cs

```diff
@@ -152,21 +186,96 @@ public TransactionInfo SpendFounderStage(ProjectInfo projectInfo, IEnumerable<st
+        return new TransactionInfo { Transaction = finalTrx, TransactionFee = totalFee };
+    }
+
+    /// <summary>
+    /// Finds the latest (most recent) expired stage release date from all investment transactions.
+    /// This is needed for Fund/Subscribe projects where different investors may have different stage schedules.
+    /// The founder can only spend a stage after ALL investors' stages have expired.
+    /// </summary>
+    private DateTime FindLatestExpiredStageReleaseDate(ProjectInfo projectInfo, List<string> investmentTransactionsHex, List<int> stageNumbers)
+    {
+        var network = _networkConfiguration.GetNetwork();
+        var latestReleaseDate = DateTime.MinValue;
+
+        for (int i = 0; i < investmentTransactionsHex.Count; i++)
+        {
+            var investmentTrxHex = investmentTransactionsHex[i];
+            var stageNumber = stageNumbers[i];
+
+            try
+            {
+                // Parse the investment transaction
+                var investmentTrx = network.CreateTransaction(investmentTrxHex);
+
+                // Create funding parameters from the transaction to get investment start date and pattern
+                var fundingParameters = FundingParameters.CreateFromTransaction(projectInfo, network.CreateTransaction(investmentTrxHex));
+
+                // Validate that we have the required data for dynamic stages
+                if (!fundingParameters.InvestmentStartDate.HasValue)
+                {
+                    _logger.LogWarning($"Investment transaction {investmentTrx.GetHash()} does not have an InvestmentStartDate. Skipping.");
+                    continue;
+                }
+
+                if (projectInfo.DynamicStagePatterns == null || !projectInfo.DynamicStagePatterns.Any())
+                {
+                    throw new InvalidOperationException("Fund/Subscribe projects must have at least one DynamicStagePattern");
+                }
+
+                if (fundingParameters.PatternIndex >= projectInfo.DynamicStagePatterns.Count)
+                {
+                    _logger.LogWarning($"Investment transaction {investmentTrx.GetHash()} has invalid PatternIndex {fundingParameters.PatternIndex}. Skipping.");
+                    continue;
+                }
+
+                // Get the pattern for this investment
+                var pattern = projectInfo.DynamicStagePatterns[fundingParameters.PatternIndex];
+
+                // Calculate the release date for the specified stage (stageNumber is 1-based, so subtract 1)
+                var stageIndex = stageNumber - 1;
+                var stageReleaseDate = DynamicStageCalculator.CalculateDynamicStageReleaseDate(
+                    fundingParameters.InvestmentStartDate.Value,
+                    pattern,
+                    stageIndex);
+
+                // Track the latest (most recent) release date
+                if (stageReleaseDate > latestReleaseDate)
```

Comment:
I don't really understand the logic here, won't the last stage always be the stage selected?

Reply:
we must find the latest date on all spending sarges for the lock time on the trx